syntax = "proto3";

package post;

service PostService {
  rpc CreatePost (CreatePostRequest) returns (CreatePostResponse);
  rpc ListPostsByUser (ListPostsByUserRequest) returns (ListPostsByUserResponse);
}

message Post {
  int64 id = 1;
  int64 user_id = 2;
  string content = 3;
  string created_at = 4;
}

message CreatePostRequest {
  int64 user_id = 1;
  string content = 2;
}

message CreatePostResponse {
  Post post = 1;
}

message ListPostsByUserRequest {
  int64 user_id = 1;
}

message ListPostsByUserResponse {
  repeated Post posts = 1;
}
